i18n tilføjet (#39)
SW705E19/report: sections/implementation/i18n.tex

\section{Translations}
To have a scalable application it is necessary to easily be able to translate the text to different languages.
For this purpose we have used i18next, which is an internationalization framework for React and React native \cite{react-i18next}.

> Instead of hardcoding the text on every page, we use the component withTranslations and gets the function t.  

Reviewer: Det er et internationalization for alle nodebaserede services, der er bare fx. en react extension til det

An example can be seen in \autoref{withTranslationsExample}. 

\begin{lstlisting}[caption={Example of withTranslation in use}, captionpos=b, label={withTranslationsExample}]
    import { withTranslation } from "react-i18next";

    function Example(){
        return(
            <h1> {t(firstname)} <h1/>
        );
    }

>     export default withTranslation()(Example)

Reviewer: hvorfor ikke vores egen kode tho

\section{Translations}
To have a scalable application it is necessary to easily be able to translate the text to different languages.
For this purpose we have used i18next, which is an internationalization framework for JavaScript.

> It is supported for many different frameworks like React, React Native, .net, Angular and many more \cite{react-i18next}.

Reviewer 2: React native er vist ikke supported
https://www.i18next.com/overview/supported-frameworks

> Instead of hardcoding the text on every page, we use the component withTranslations and gets the function t.  

Reviewer 2: ```suggestion
Instead of hardcoding the text on every page, we wrap the export of the component in the function withTranslations which injects the function t into the props of the component.  
```

> Instead of hardcoding the text on every page, we use the component withTranslations and gets the function t.  

Reviewer 3: ```suggestion
Instead of hardcoding the text on every page, we use the component \texttt{withTranslations} to get the function \texttt{t] for translation purposes.  
```

\begin{lstlisting}[caption={Translated header when registering as a user.}, captionpos=b, label={withTranslationsUserForm}]
    import { withTranslation } from "react-i18next";

    function UserForm(props){
        const { t, classes } = props;

        return (
            <Container maxWidth="sm">
               <div className={classes.paper}>
                  <Typography align="center" variant="h4">
                      {t('registerasauser')}
                  </Typography>
               ...
        );
    }

    export default withTranslation()(withStyles(styles)(UserForm))
\end{lstlisting}
\noindent

> In \autoref{withTranslationsUserForm} is the translation for the header on the register page.

Reviewer 2: ```suggestion
The translation of the the title on the register page can be seen in \autoref{withTranslationsUserForm}.
```

> In \autoref{withTranslationsUserForm} is the translation for the header on the register page.

Reviewer 3: ```suggestion
\autoref{withTranslationsUserForm} shows the translation for the header on the register page.
```

> The \texttt{withTranslation()} component will then look in the translation files to find the correct translation for "registerasauser". 

Reviewer 2: ```suggestion
The \texttt{withTranslation()} function injects the function t into the props of the component. When called the function t will look up the passed key in the translation files to find the correct translation for "registerasauser". 
```

> The \texttt{withTranslation()} component will then look in the translation files to find the correct translation for "registerasauser". 

Reviewer 3: ```suggestion
The \texttt{withTranslation()} component will then look in the translation files to find the correct translation for \textit{"registerasauser"}. 
```

> The \texttt{withTranslation()} function injects the function t into the props of the component. When called the function t will look up the passed key in the translation files to find the correct translation for  \textit{"registerasauser"}. 

Reviewer 3: ```suggestion
The \texttt{withTranslation()} function injects the function t into the props of the component.
When called, the function t will look up the passed key in the translation files to find the correct translation for  \textit{"registerasauser"}. 
```

A screenshot of the files can be seen on \autoref{fig:translationfiles}.
\begin{figure}
    \centering
    \includegraphics[scale=0.5]{figures/translations.PNG}
    \caption{The Danish and English translation files}
    \label{fig:translationfiles}
\end{figure}
\noindent

> To add an additional language all that needs to be done is to make a new file and translate all the lines. 

Reviewer 2: ```suggestion
To add an additional language a new translation file needs to be created. All the lines that are present in the other files then need to be translated to the desired language. 
```

> To add an additional language a new translation file needs to be created. All the lines that are present in the other files then need to be translated to the desired language. 

Reviewer 3: ```suggestion
To add an additional language a new translation file needs to be created.
All the lines that are present in the other files then need to be translated to the desired language. 
```
